### ISSUE TITLE
`Video.Assets.create` doesn't just return asset anymore

### ISSUE THREAD

**Issue by hellovietduc** [muxinc/mux-node-sdk#134]
I have the following code:

```javascript
const asset = await Video.Assets.create({
  input: videoUrl,
  playback_policy: 'public',
})
console.info(asset)
```

While the type definition clearly indicates that `Video.Assets.create` would resolve to an asset object that looks like this:

```json
{
  "status": "preparing",
  "playback_ids": [
    {
      "policy": "public",
      "id": "OHWwY9L00OoSNAQ7MrtDH00Gal4S01z7wyeszVLIThmbh00"
    }
  ],
  "mp4_support": "none",
  "master_access": "none",
  "id": "H4I300fePrcZs6lJ009m2vyeiLtojQ7u7FESR3CRf3tZY",
  "created_at": "1632108478"
}
```

Now it resolves to an object that looks like a HTTP response:

```json
{
  "status": 201,
  "statusText": "Created",
  "headers": {
    ...
  },
  "config": {
    ...
  },
  "request": {
    ...
  },
  "data": {
    "data": {
      "status": "preparing",
      "playback_ids": [
        {
          "policy": "public",
          "id": "OHWwY9L00OoSNAQ7MrtDH00Gal4S01z7wyeszVLIThmbh00"
        }
      ],
      "mp4_support": "none",
      "master_access": "none",
      "id": "H4I300fePrcZs6lJ009m2vyeiLtojQ7u7FESR3CRf3tZY",
      "created_at": "1632108478"
    }
  }
}
```

I'm using `@mux/mux-node` v3.2.3.

**Comment by dylanjha** [comment]
Hi @hellovietduc 

That looks suspiciously like #130 that @michielswaanen opened a couple weeks ago. We weren't able to reproduce it, I'm not sure if they ever got to the bottom of it.

Can you please:

1. Provide a reproducible case [this script that we used to debug in #130](https://gist.github.com/dylanjha/30f07784a4e646008a2898b0ece9ed96) would be a good start (go ahead and change `Video.Uploads.create` to `Video.Assets.create`)
2. Can you expand the full response that you pasted up there^ (include as much as you can, but definitely DON'T include your API key/secret)

**Comment by michielswaanen** [comment]
We didn't find the source yet (we haven't looked further into either). 

All I can tell is that we are using a custom Axios client for most of our requests. Maybe that has something to do with it?

**Comment by hellovietduc** [comment]
I might have found it. Yes, it has something to do with `axios`. I'm using Yarn workspaces in my project. In workspace `request` I install `axios@0.21.2` (absolute version). `@mux/mux-node` is installed in workspace `transcode`. I found out that the SDK uses `axios@0.21.2` instead of installing a newer version of `axios`.

**Comment by dylanjha** [comment]
@hellovietduc @michielswaanen

A custom axios client definitely can be the issue -- the base constructor [sets up the axios client like this](https://github.com/muxinc/mux-node-sdk/blob/76b7ce93bbbcdebae6fc2313f9b401522f9cf0ba/src/base.js#L37-L72)

Particularly notice the response interceptor. 

My question is -- how are you configuring your application to override this SDK's base level HTTP functionality? Can you provide sample code that does that?

**Comment by michielswaanen** [comment]
As far as I know we're not overriding any configuration, otherwise we would have found the issue pretty easily..

I'll keep you updated.

**Comment by hellovietduc** [comment]
I also don't override any configuration.

**Comment by hellovietduc** [comment]
I can reproduce the error when forcing the SDK to use `axios@0.21.2` like this:

```javascript
/*
  yarn init -y
  yarn add axios@0.21.2 @mux/mux-node@3.2.3
  rm -rf node_modules/@mux/mux-node/node_modules/axios
  ln -s node_modules/axios node_modules/@mux/mux-node/node_modules/axios
  MUX_TOKEN_ID=xxxxx MUX_TOKEN_SECRET=xxxx node ./mux.js
*/

const Mux = require('@mux/mux-node');
const pkg = require('@mux/mux-node/package.json');

console.log('Version', pkg.version);

const { Video } = new Mux();
const videoUrl = 'https://file-examples-com.github.io/uploads/2017/04/file_example_MP4_480_1_5MG.mp4';

Video.Assets.create({ input: videoUrl, playback_policy: 'public' })
  .then(console.log)
  .catch(console.error);

```

**Comment by dylanjha** [comment]
Ahh, that makes sense with the different axios versions

I think it goes back to this change:

* https://github.com/axios/axios/pull/2391
* https://github.com/axios/axios/blob/master/CHANGELOG.md#0191-jan-7-2020

Will work on a patch

**Comment by dylanjha** [comment]
Okay - so those issues were not relevant^... but I did get to the bottom of it

The issue is specific to axios `0.21.2` because [there was a regression in that version]. The regression was quickly patched it in `0.21.3`. The regression is that our response interceptor was not getting registered. I was able to replicate this and then confirm that `0.21.3` and later do not have the issue

I'm going to go ahead and update our dep from `^0.21.1` to the latest `^0.21.4` and cut a new patch version of mux-node

**Comment by dylanjha** [comment]
Fixed in [v3.3.1](https://github.com/muxinc/mux-node-sdk/releases/tag/v3.3.1)

**Comment by hellovietduc** [comment]
Thank you!
